node/network/bitfield-distribution/src/lib.rs

```diff
@@ -0,0 +1,684 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! The bitfield distribution
+//!
+//! In case this node is a validator, gossips its own signed availability bitfield
+//! for a particular relay parent.
+//! Independently of that, gossips on received messages from peers to other interested peers.
+
+use codec::{Decode, Encode};
+use futures::{channel::oneshot, FutureExt};
+
+use node_primitives::{ProtocolId, View};
+
+use log::{debug, info, trace, warn};
+use polkadot_node_subsystem::messages::*;
+use polkadot_node_subsystem::{
+	FromOverseer, OverseerSignal, SpawnedSubsystem, Subsystem, SubsystemContext, SubsystemResult,
+};
+use polkadot_primitives::v0::{SigningContext, ValidatorId};
```

Comment:
Should not be using any `v0` primitives, only `v1` - types by the same name are exposed.